### PR TITLE
Feature: enable dark mode

### PR DESCRIPTION
**Pull request by alexsan9051**
Added Dark Mode While Timer is Running mode.

### CODE REVIEW

File: src/lib/atom.ts [hunk 1]
```diff
@@ -95,6 +95,11 @@ const getUpdatedTimeAtom = atom((get) => {
 
 const selectedTodoIdAtom = atom(fetchSelectedTodoIdFromStorage() ?? '');
 
+// Theme settings
```
Comment (teoh4770) [inline src/lib/atom.ts:98]: using object is great here

File: src/components/Setting/ThemeSettingInputs.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const handleChange = () => {
+        setThemeSetting({
+            ...themeSetting,
+            darkModeWhenRunning: !themeSetting.darkModeWhenRunning,
+        });
+    };
+
+    return (
+        <div className="theme-setting">
+            <h3 className="setting-title capita py-4 font-semibold text-gray-400">
```
Comment (teoh4770) [inline src/components/Setting/ThemeSettingInputs.tsx:20]:
```suggestion
            <h3 className="setting-title capita py-4 font-semibold text-gray-400">
```
my bad, "capita"  -> "capitalize"

File: src/App.tsx [hunk 3]
```diff
@@ -2,16 +2,26 @@ import { Button } from './components/ui';
 import { Setting, Timer, Todos, Summary } from './components';
 import { useTimer } from './hooks';
 import { runTour } from './lib';
+import { useAtomValue } from 'jotai';
+import { themeSettingsAtom } from './lib/atom';
```
Comment (teoh4770) [inline src/App.tsx:6]:
```suggestion
import { themeSettingsAtom } from './atom';
```
can you try that

File: src/App.tsx [hunk 4]
```diff
@@ -2,16 +2,26 @@ import { Button } from './components/ui';
...
 
 const App = () => {
     const timer = useTimer();
     const resetTimer = timer.actions.resetTimer;
+    const themeSettings = useAtomValue(themeSettingsAtom);
+    const isDarkModeWhenRunning = themeSettings.darkModeWhenRunning;
+
+    // Determine if the timer is running
```
Comment (teoh4770) [inline src/App.tsx:14]: no need the comment here, since it is pretty clear already

Comment (teoh4770) [inline src/App.tsx:14]:
```suggestion
    
```

**Comment by teoh4770** [comment]
Screenshot 2024-04-27 at 11 18 26 PM
Screenshot 2024-04-27 at 11 21 15 PM

Instead of adding extra props to the components, you can instead:
1. toggle the "hiddenClass" at the main container (first picture)
2. add a visible class to the components that you wanna keep visibility (second picture,  I want to keep the timer visible, so I add a visible class)
3. to style the component that has visible class when dark mode is toggled, you can do something like

```css
// this is exception class
.hiddenClass .visible .timer-section {
	...
}
```

File: src/App.tsx (hunk 3 above)
Comment (alexsan9051) [reply src/App.tsx:6]: _"Cannot find module './atom' or its corresponding type declarations"_

**Comment by alexsan9051** [comment]
> Instead of adding extra props to the components, you can instead do the following:
> 
>     1. toggle the "hiddenClass" at the main container (first picture)
> 
>     2. add a visible class to the components that you wanna keep visibility (second picture,  I want to keep the timer visible, so I add a visible class)
> 
>     3. to style the component that has visible class when dark mode is toggled, you can do something like
> 
> 
> ```css
> // this is exception class
> .hiddenClass .visible .timer-section {
> 	...
> }
> ```

This approach would get rid of the transition between visible and invisible.

Also, even if the component with the class "visible" is visible, its content would still be hidden due to being within the parent <main> with the "invisible" class, Haven't been able to make certain parts of a component visible when hiddenClass is applied to <main>. Not sure how we can avoid extra props.

File: src/App.tsx (hunk 3 above)
Comment (teoh4770) [reply src/App.tsx:7]: in that case, keep it like that first

**Review by teoh4770 — APPROVED** [review]
Alright, i have cleaned up some of the code. Approve!

**Pull request merged**